ADRs/0011-Python-Path-Resolution.md

```diff
@@ -0,0 +1,631 @@
+        > rsa | 4.0
+        > scikit-learn | 0.22.2.post1
+        > scipy | 1.4.1
+        > setupext-janitor | 1.1.2
+        > six | 1.15.0
+        > tensorboard | 2.1.1
+        > tensorflow | 2.1.0
+        > tensorflow-estimator | 2.1.0
+        > termcolor | 1.1.0
+        > toml | 0.10.0
+        > torch | 1.4.0+cpu
+        > torchvision | 0.5.0+cpu
+        > tqdm | 4.45.0
+        > twine | 3.1.1
+        > typed-ast | 1.4.0
+        > urllib3 | 1.25.9
+        > webencodings | 0.5.1
+        > Werkzeug | 1.0.0
+        > wrapt | 1.11.2
+        > zipp | 3.1.0
+.
+.
+.
+
+-   id: 3   
+    path: C:\Users\shams\AppData\Local\Microsoft\WindowsApps\python.exe                        
+    version: 3.6.1
+    .
+    .
+    .
+
+-----------------------------------------------------------------------------------------------------
+
+.
+.
+.
+
+```
+
+### Example Workflow
+Here is an example workflow that involves the actual API code snippets:
+
+#### 1. For Regular Python Install
+Assuming `konduit pythonpaths`, gives us the following information on a particular system for regular python installs:
+
+```text
+konduit-pc ~ % konduit pythonpaths
+
+-------------------------------------------PYTHON INSTALLS-------------------------------------------
+.
+.
+.
+
+-   id: 4
+    path: C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\python.exe
+    version: 3.7.8
+    packages: 
+    .
+    .
+    .
+
+.
+.
+.
+```
+
+Now we want to specify the python install with `id = 4` into the PythonStep configuration in Java. The code would look similar to: 
+
+```java
+PythonConfig pythonConfig = PythonConfig.builder()
+        .pythonType(PythonConfig.PythonType.Python)
+        .pythonPath("4")
+        .build();
+```
+
+#### 2. For Conda Install
+Assuming `konduit pythonpaths`, gives us the following information on a particular system for conda installs:
+
+```text
+konduit-pc ~ % konduit pythonpaths
+
+.
+.
+.
+
+--------------------------------------------CONDA INSTALLS-------------------------------------------
+-   id: 1
+    path: C:\Users\shams\miniconda3\Scripts\conda.exe
+    version: 4.8.4
+    ----------------------------------------ENVIRONMENTS---------------------------------------------
+    -   name: base
+        path: C:\Users\shams\miniconda3
+        version: 3.6.1
+    
+    -   name: py37
+        path: C:\Users\shams\miniconda3\envs\py37
+        version: 3.7.7
+    
+    -   name: py38
+        path: C:\Users\shams\miniconda3\envs\py38
+        version: 3.8.1
+    -------------------------------------------------------------------------------------------------
+    
+.
+.
+.
+```
+
+Now we want to specify the conda install with `id = 1` and the conda environment `py37` into the PythonStep configuration in Java. The code would look similar to: 
+
+```java
+PythonConfig pythonConfig = PythonConfig.builder()
+        .pythonType(PythonConfig.PythonType.Conda)
+        .pythonPath("1")
```

Comment:
We might want to do an object oriented/inherited approach for python paths/types to allow us to add specific variables for different environments. We risk the same kind of sprawl we used to have before otherwise. We will probably need to add a lot of specific variables for conda.